src/timepicker/timepicker.js

```diff
@@ -194,23 +210,26 @@ class TimePicker extends React.Component<TimePickerPropsT, TimePickerStateT> {
     excludeOptions,
     newProps,
   ) => {
-    const result = stringToOptions(filterValue, this.props.format);
+    const result = this.stringToOptions(filterValue, this.props.format);
     if (result.length) {
       return result;
     }
     return filterOptions(options, filterValue, excludeOptions, newProps);
   };
 
-  buildSelectedOption(value: Date, format: '12' | '24' = '12') {
-    const secs = dateToSeconds(value);
+  buildSelectedOption: (T, ?'12' | '24') => OptionT = (
+    value,
+    format = '12',
+  ) => {
+    const secs = this.dateHelpers.dateToSeconds(value);
     return {
       id: secs,
-      label: secondsToLabel(secs, format),
+      label: this.secondsToLabel(secs, format || '12'),
```

Comment:
Is the `|| '12'` required when it has a default value already?

Reply:
It is yah, unfortunately.  The long term solution to this is probably to not pass it around at all and just use the format prop directly in these functions, but there was some old behavior that that could have potentially broken, so I decided to wait on that.